pkg/issuer/acme/dns/route53/route53_test.go

```diff
@@ -45,44 +43,37 @@ func makeRoute53Provider(ts *httptest.Server) *DNSProvider {
 	return &DNSProvider{client: client}
 }
 
-func TestCredentialsFromEnv(t *testing.T) {
+func TestAmbientCredentialsFromEnv(t *testing.T) {
 	os.Setenv("AWS_ACCESS_KEY_ID", "123")
 	os.Setenv("AWS_SECRET_ACCESS_KEY", "123")
 	os.Setenv("AWS_REGION", "us-east-1")
+	defer restoreRoute53Env()
 
-	config := &aws.Config{
-		CredentialsChainVerboseErrors: aws.Bool(true),
-	}
+	provider, err := NewDNSProvider("", "", "", "", true)
+	assert.NoError(t, err, "Expected no error constructing DNSProvider")
 
-	sess := session.New(config)
-	_, err := sess.Config.Credentials.Get()
+	_, err = provider.client.Config.Credentials.Get()
 	assert.NoError(t, err, "Expected credentials to be set from environment")
-
-	restoreRoute53Env()
 }
 
-func TestRegionFromEnv(t *testing.T) {
+func TestNoCredentialsFromEnv(t *testing.T) {
+	os.Setenv("AWS_ACCESS_KEY_ID", "123")
+	os.Setenv("AWS_SECRET_ACCESS_KEY", "123")
 	os.Setenv("AWS_REGION", "us-east-1")
+	defer restoreRoute53Env()
 
-	sess := session.New(aws.NewConfig())
-	assert.Equal(t, "us-east-1", *sess.Config.Region, "Expected Region to be set from environment")
-
-	restoreRoute53Env()
+	_, err := NewDNSProvider("", "", "", "", false)
+	assert.Error(t, err, "Expected error constructing DNSProvider with no credentials and not ambient")
 }
 
-func TestHostedZoneIDFromEnv(t *testing.T) {
-	const testZoneID = "testzoneid"
-
+func TestAmbientRegionFromEnv(t *testing.T) {
+	os.Setenv("AWS_REGION", "us-east-1")
```

Comment:
So it seems like if the 'region' field is not set in the Issuer resource, we use the environment to determine which to use (this var).

I guess this happens as part of credential loading in the default chain? & additionally, if AWS_REGION is set, but ambient credentials are *disabled*, will the region still be used? It's not such a big security concern, but it could cause strange behaviour for users and mask the fact they've not set the region explicitly in the YAML.

Reply:
Region handling is distinct from credential chains. The two sources that will be consulted for region are that environment variable, and the 'shared config' (the ini file at `~/.aws/config`). Notably, ec2 metadata isn't consulted.

The current implementation of the ambient option does not impact those settings. I didn't really think about that, so I left it as it was before.

The region thing is a bit silly here since route53 is a global service, so the region only really matters if you want to reduce api latency by hitting a nearer server, or if you're using a different aws partition (like aws china or govcloud).

I'll go ahead and make it respect the ambient config for now, but I think the above may be an argument for defaulting regardless it in the future.

Reply:
Added a commit to do the above, and updated those unit tests.